bitwarden-desktop: 2024.12.1 -> 2025.1.1

Saeverix opened the pull request:
Diff: https://github.com/bitwarden/clients/compare/desktop-v2024.12.1...desktop-v2025.1.1

Changelog: https://github.com/bitwarden/clients/releases/tag/desktop-v2025.1.1

* Please also backport

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Saeverix committed:
bitwarden-desktop: 2024.12.1 -> 2025.1.1

Diff: https://github.com/bitwarden/clients/compare/desktop-v2024.12.1...desktop-v2025.1.1

Changelog: https://github.com/bitwarden/clients/releases/tag/desktop-v2025.1.1

Saeverix commented:
Might make #373177 obsolete

Saeverix commented:
@nyawox thank you for the tips. I see that this is not a simple update then. Currently the update is even impossible, since `electron_33` isn't even available yet on NixOS. Guess I will have to put this one in the fridge for now.

nyawox commented:
oh yea my bad, since `os-biometrics-linux.service.ts` rename was merged to nightly https://github.com/bitwarden/clients/tree/main/apps/desktop/src/key-management/biometrics, i automatically assumed it will make to stable release.

Saeverix commented:
Looks like it's still `biometric.unix.main.ts` in https://github.com/bitwarden/clients/tree/desktop-v2025.1.0/apps/desktop/src/key-management/biometrics, so we are good on that.

I think it's best to let [#347350](https://redirect.github.com/NixOS/nixpkgs/issues/347350) and [#371479](https://redirect.github.com/NixOS/nixpkgs/issues/371479) be fixed in their own separate pull requests, since it's not really related to the update that I am doing.

nyawox commented:
opened a [separate pr](https://github.com/NixOS/nixpkgs/pull/374068) 👍🏻

Saeverix commented:
I suspect @amarshall is not very active recently. Since the linked PR #73177 is open with no response of him for a week. Maybe @ddogfoodd can check this out, since you also participated in #73177?

amarshall reviewed (changes requested):
Thanks! Just one comment.

Saeverix commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 374021`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>bitwarden-desktop</li>
  </ul>
</details>

Saeverix commented on pkgs/by-name/bi/bitwarden-desktop/package.nix:
```diff
@@ -3,13 +3,13 @@
   gtk3,
   jq,
-  libsecret,
+  oo7,
```
I assume when you say "keep it alphabetized", there is a rule for `lib` being an exception and always on top?

Saeverix commented:
@amarshall and @ddogfoodd I have resolved the comment. I also updated to 2025.1.1 while we are at it. Rebuilt successful without `oo7`, `glib` and `gtk3`, so I removed `buildInputs = [];` completely.

Saeverix commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 374021`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>bitwarden-desktop</li>
  </ul>
</details>

Saeverix commented:
I had some unnecessary merge conflicts to resolve, because #373177 was merged by @ck3d even though I mention this PR in there. But anyway, branch is up to date again and this one can be merged now. Also make sure to add the backport label after merging.

Saeverix commented:
@amarshall a friendly reminder

amarshall reviewed (approved):
Built on x86_64-linux and performed usual manual tests. Thanks!

Saeverix commented:
Thanks @amarshall, I assume you are also the one to merge the branch? Since I don't have the required permssion. Or should someone else merge the branch?

talyz reviewed (approved):
Works well for me. Tested by running `nix run github:Saeverix/nixpkgs/update-bitwarden-desktop#bitwarden-desktop` on `x86_64-linux`.

Thanks!

Saeverix commented:
@talyz (and or @ddogfoodd) I see the automated backport failed for some reason. Is that something I can fix, or should one of you do that since I don't have the required permissions?
If I can help, then I would please have some instructions on how to do it right.
